tests/neg/i10901.scala

```diff
@@ -43,8 +43,8 @@ object BugExp4Point2D {
     val y = DoubleT
 
     val pos1: Point2D[Int,Double] = x º y       // error
-    val pos2: Point2D[Int,Double] = 100 º 200.1 // ok
-    val pos3: Point2D[Int,Double] = 101 º y     // ok
+//  val pos2: Point2D[Int,Double] = 100 º 200.1
```

Comment:
can we justify why these should now be broken? I tried the following change but it still did not type check.
It seems to be a tricky position where the compiler can not handle inference well for context parameters after the extension clause.
```scala
    extension [T1:Numeric](x: T1)

      // N - N
      @targetName("point2DConstant")
      def º[T2:Numeric](y: T2): Point2D[T1,T2] = ???


      // N - C
      @targetName("point2DConstantData")
      def º[T2:Numeric](y: ColumnType[T2]): Point2D[T1,T2] = ???



    extension [T1:Numeric](x: ColumnType[T1])
      // C - C
      @targetName("point2DData")
      def º[T2:Numeric](y: ColumnType[T2]): Point2D[T1,T2] = ???

      // C - N
      @targetName("point2DDataConstant")
      def º[T2:Numeric](y: T2): Point2D[T1,T2] = ???
```

Reply:
Well I don't know, overloads don't help for sure. Without these, it works, i.e. just C-C and N-N for instance. In addition, I had to comment lines 46-47 instead of reporting an error, because the latter is undeterministic (it changes at each recompile):
```scala
    val pos2: Point2D[Int,Double] = 100 º 200.1
                                    ^^^^^
value º is not a member of Int.
An extension method was tried, but could not be fully constructed:

    BugExp4Point2D.dsl.º[Int, T2](100)(scala.math.Numeric.IntIsIntegral, 
      /* ambiguous: both object FloatIsFractional in object Numeric and object CharIsIntegral in object Numeric match type Numeric[T2] */
        summon[Numeric[T2]]
    )    failed with

        ambiguous implicit arguments: both object FloatIsFractional in object Numeric and object CharIsIntegral in object Numeric match type Numeric[T2] of an implicit parameter of method º in object dsl
```

Reply:
its probably worth opening an issue for the manual desugaring not being found by the compiler

Reply:
https://github.com/lampepfl/dotty/issues/11713